hack/jenkins/e2e.sh

```diff
@@ -111,6 +111,7 @@ GKE_REQUIRED_SKIP_TESTS=(
     "MasterCerts"
     "Shell"
     "Daemon\sset"
+    "Deployment"
```

Comment:
Please fix the comment above, as it is now inaccurate.  The reason this test is excluded is not because it requires SSH access, but because it requires access to experimental API's. 
Also, please file an issue to remove this filter as soon as Deployments move out of experimental (and ideally a comment in the code that would change when it moves out of experimental, wherever that is).